code/src/App.js

```diff
@@ -1,9 +1,116 @@
-import React from 'react'
+import React,{ useState, useEffect } from 'react';
+
+import Header from './Component/Header';
+import ThoughtSubmitForm from 'Component/ThoughtSubmitForm';
+import MessageList from 'Component/MessageList';
+import { API_URL } from './constants';
+import './index.css';
+
+//there are 3 erros found in the response relating to empty, too short or too long messages.
+const THOUGHT_ERRORS = [
+  { kind: 'required', message: 'Please type in your happy thought' },
+  { kind: 'minlength', message: 'Message should be longer than 5 characters' },
+  { kind: 'maxlength', message: 'Oops! Can you keep the message shorter than 140 characters, please.' }
+];
 
 export const App = () => {
+  const [thought, setThought] = useState(""); //the thought state needs to be here in the App.js instead in the child component to handle the erro update while hitting submit button.
+  const [thoughts, setThoughts] = useState([]);
```

Comment:
these state variables are quite similar in the naming.. Maybe it could be more clear.